qpid/cpp/src/qpid/broker/amqp/Outgoing.cpp

```diff
@@ -142,11 +145,10 @@ void OutgoingFromQueue::handle(pn_delivery_t* delivery)
                 if (preAcquires()) {
                     //TODO: handle message-annotations
                     if (pn_disposition_is_undeliverable(pn_delivery_remote(delivery))) {
-                        //treat undeliverable here as rejection
-                        queue->reject(r.cursor);
-                    } else {
-                        queue->release(r.cursor, pn_disposition_is_failed(pn_delivery_remote(delivery)));
+                        undeliverableMessages.add(r.msg.getSequence());
                     }
+
+                    queue->release(r.cursor, pn_disposition_is_failed(pn_delivery_remote(delivery)));
```

Comment:
That's a fair point. It wouldn't actually be a regression. Let's leave any queue options as a separate concern for now then.